src/site/content/en/learn/css/inheritance/index.md

```diff
@@ -0,0 +1,341 @@
+---
+title: Inheritance
+description:
+authors:
+  - andybell
+date: 2021-04-02
+tags:
+  - css
+---
+
+# Inheritance
+
+Say you just wrote some CSS to make elements look like a button.
+
+```html
+<a href="http://example.com" class="my-button">I am a button link</a>
+```
+
+```css
+.my-button {
+  display: inline-block;
+  padding: 1rem 2rem;
+  text-decoration: none;
+  background: #eabdb3;
+  font: inherit;
+  text-align: center;
+}
+```
+
+You then add a link element to an article of content,
+with a `class` value of `.my-button`. However there's an issue,
+the text is not the color that you expected it to be. How did this happen?
+
+Some CSS properties inherit if you don't specify a value for them.
+In the case of this button, it **inherited** the `color` from this CSS:
+
+```css
+article a {
+  color: #7c3626;
+}
+```
+
+In this lesson you'll learn why that happens and
+how inheritance is a powerful feature to help you write less CSS.
+
+<figure class="w-figure">
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'zYNGEbg'
+} %}
+</figure>
+
+## Inheritance flow
+
+Let's take a look at how inheritance works,
+using this snippet of HTML:
+
+```html
+<html>
+  <body>
+    <article>
+      <p>Lorem ipsum dolor sit amet.</p>
+    </article>
+  </body>
+</html>
+```
+
+The root element (`<html>`) won't inherit anything because it is the first element in the document.
+Add some CSS on the HTML element,
+and it starts to cascade down the document.
+
+```css
+html {
+  color: lightslategray;
+}
+```
+
+<figure class="w-figure">
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'JjEKgBX'
+} %}
+</figure>
+
+The `color` property is inheritable by other elements.
+The `html` element has `color: lightslategray`,
+therefore all elements that can inherit color will now have a color of `lightslategray`.
+
+```css
+body {
+  font-size: 1.2em;
+}
+```
+
+<figure class="w-figure">
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'VwPLrLP'
+} %}
+</figure>
+
+{% Aside %}
+Because this demo sets the font size on the `body` element,
+the `html` element will still have the initial font size set by the browser (user agent stylesheet),
+but the `article` and `p` will inherit the font size declared by the `body`.
+This is because inheritance only cascades downwards.
+{% endAside %}
+
+```css
+p {
+  font-style: italic;
+}
+```
+
+Only the `<p>` will have italic text because it's the deepest nested element.
```

Comment:
The `<p>` is the only element in this demo. Can we add another one to show this a little more clearly? Currently all of the demos look similar, while trying to show different things.